presto-orc/src/main/java/com/facebook/presto/orc/reader/SelectiveStreamReaders.java

```diff
@@ -138,6 +142,44 @@ public static SelectiveStreamReader createStreamReader(
         }
     }
 
+    private static SelectiveStreamReader getDoubleSelectiveStreamReader(StreamDescriptor streamDescriptor, Map<Subfield, TupleDomainFilter> filters, Optional<Type> outputType,
+            OrcAggregatedMemoryContext systemMemoryContext, boolean orcUseVectorFilter, OrcTypeKind type)
+    {
+        if (orcUseVectorFilter) {
+            try {
+                Class doubleSelectiveStreamVectorReaderClass = Class.forName("com.facebook.presto.orc.reader.vector.DoubleSelectiveStreamVectorReader",
+                        false, SelectiveStreamReaders.class.getClassLoader());
+                // Ensure Memory Segment class is availabe for use
+                Class memorySegmentClass = Class.forName("java.lang.foreign.MemorySegment");
+                // Ensure DoubleVector and VectoMask classes included in the jdk
+                Class doubleVector = Class.forName("jdk.incubator.vector.DoubleVector");
+                Class vectorMask = Class.forName("jdk.incubator.vector.VectorMask");
+                // Ensure compress api is available used in vector implementation
+                Class[] methodParameterArgs = new Class[1];
+                methodParameterArgs[0] = vectorMask;
+                Method commpressApi = doubleVector.getDeclaredMethod("compress", methodParameterArgs);
+
+                // create the vector class during runtime if the vector path is chosen for execution else fallback to the old execution flow
+                Class[] constructorArguments = new Class[4];
+                constructorArguments[0] = StreamDescriptor.class;
+                constructorArguments[1] = Optional.class;
+                constructorArguments[2] = boolean.class;
+                constructorArguments[3] = OrcLocalMemoryContext.class;
+                Constructor constructor = doubleSelectiveStreamVectorReaderClass.getDeclaredConstructor(constructorArguments);
+                return (SelectiveStreamReader) constructor.newInstance(streamDescriptor,
+                        getOptionalOnlyFilter(type, filters),
+                        outputType.isPresent(),
+                        systemMemoryContext.newOrcLocalMemoryContext(SelectiveStreamReaders.class.getSimpleName()));
+            }
+            catch (ClassNotFoundException | NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
```

Comment:
we should fail loudly here.

Reply:
Can you please help me know ? And BTW there is old flow available if the incubator classes are not available during runtime.

Reply:
I mean to say instead of falling back if the classes aren't available, it would be better to fail the query so people know they have misconfigured presto.